app/models/Entity.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace app\models;
+
+use app\core\Database;
+
+abstract class Entity {
```

Comment:
I'd consider using an Interface here: https://www.php.net/manual/en/language.oop5.interfaces.php

Reply:
Thanks for the tip, Java has the same rules for interface, I am simply using the abstract class to save my effort of instantiating the Database class in the child class lol. I would refactor my code to reflect the interface implementation.

Reply:
But I assume it's indeed best practice to use interface in this situation, because the entity is only providing a bunch of CRUD function declarations. Also, this would provide flexibility of my child class to inherit other parent classes.